jupyterhub/templates/hub/configmap.yaml

```diff
@@ -5,312 +5,21 @@ metadata:
-  {{- end }}
-  {{- if include "jupyterhub.userPodAntiAffinityRequired" . }}
-  singleuser.pod-anti-affinity-required: |
-    {{- include "jupyterhub.userPodAntiAffinityRequired" . | nindent 4 }}
-  {{- end }}
-  {{- if include "jupyterhub.userPodAntiAffinityPreferred" . }}
-  singleuser.pod-anti-affinity-preferred: |
-    {{- include "jupyterhub.userPodAntiAffinityPreferred" . | nindent 4 }}
-  {{- end }}
-
-  {{- if .Values.scheduling.userScheduler.enabled }}
-  singleuser.scheduler-name: "{{ .Release.Name }}-user-scheduler"
-  {{- end }}
-  {{- if .Values.scheduling.podPriority.enabled }}
-  singleuser.priority_class_name: "{{ .Release.Name }}-default-priority"
-  {{- end }}
-
-  {{- /* KubeSpawner */}}
-  kubespawner.common-labels: |
-    {{- $_ := merge (dict "heritageLabel" "jupyterhub") . }}
-    {{- include "jupyterhub.commonLabels" $_ | nindent 4 }}
-
-
-  {{- /* Hub */}}
-  hub.allow-named-servers: {{ .Values.hub.allowNamedServers | quote }}
-  hub.concurrent-spawn-limit: {{ .Values.hub.concurrentSpawnLimit | quote }}
-  hub.consecutive-failure-limit: {{ .Values.hub.consecutiveFailureLimit | quote }}
-  {{- if .Values.hub.activeServerLimit }}
-  hub.active-server-limit: {{ .Values.hub.activeServerLimit | quote }}
-  {{- end }}
-  {{- if .Values.hub.services }}
-  hub.services: |
-    {{- /* include hub.services *except* api_tokens */}}
-    {{- range $name, $service := .Values.hub.services }}
-    {{ $name }}:
-      {{- range $key, $value := $service }}
-      {{- if ne $key "apiToken" }}
-      {{- /* Use toJson since values can be more than just strings */}}
-      {{ $key }}: {{ toJson $value }}
-      {{- end }}
-      {{- end }}
-    {{- end }}
-  {{- end }}
-  {{- if .Values.hub.extraConfig }}
-  {{- $extraConfigType := typeOf .Values.hub.extraConfig }}
-  {{- if eq $extraConfigType "string" }}
-  hub.extra-config.default.py: |
-    {{- .Values.hub.extraConfig | nindent 4 }}
-  {{- else if eq $extraConfigType "map[string]interface {}" }}
-  {{- range $key, $value := .Values.hub.extraConfig }}
-  hub.extra-config.{{ $key }}.py: |
-    {{- $value | nindent 4 }}
-  {{- end }}
-  {{- end }}
-  {{- end }}
-  {{- if .Values.hub.extraConfigVars }}
-  hub.extra-vars: {{ toJson .Values.hub.extraConfigVars | quote }}
-  {{- end }}
-  hub.base_url: {{ .Values.hub.baseUrl | quote }}
-  {{- if eq .Values.hub.db.type "sqlite-pvc" }}
-  hub.db_url: "sqlite:///jupyterhub.sqlite"
-  {{- else if eq .Values.hub.db.type "sqlite-memory" }}
-  hub.db_url: "sqlite://"
-  {{- else }}
-  hub.db_url: {{ .Values.hub.db.url | quote }}
-  {{- end }}
-  {{- if .Values.hub.cookieSecret }}
-  token.cookie_secret: {{ .Values.hub.cookieSecret | quote }}
-  {{- end }}
-
-  {{- /* Other */}}
-  debug.enabled: {{ .Values.debug.enabled | quote }}
-  {{- range $key, $value := .Values.hub.extraConfigMap }}
-  custom.{{ $key }}: {{ toJson $value | quote }}
-  {{- end }}
+{{- $values := pick .Values "auth" "cull" "custom" "hub" "scheduling" "singleuser" }}
```

Comment:
pick `"debug"` as well

Reply:
No, that would be wrong.

Reply:
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/ef4b0f757ac3314e2679b16a66373a372df960ae/images/hub/jupyterhub_config.py#L333-L335

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/ef4b0f757ac3314e2679b16a66373a372df960ae/jupyterhub/templates/hub/configmap.yaml#L312-L313

Reply:
@minrk good idea I think would be simpler to maintain